src/Modules/General/SearchComponent.js

```diff
@@ -1,110 +1,87 @@
-                        </div> */}
-                        <div className="form-group zip-code">
-                            <label htmlFor="zip">Zip</label>
-                            <input
-                                type="text"
-                                className="form-control"
-                                id="zip"
-                                name="zip"
-                                value={zip || ''}
-                                onChange={buildData}
-                            />
+           {showAddress &&  <div className="form-group flex-grow-1" data-testid="search-street">
+                            <label htmlFor="street">Street</label>
+                             <PlacesAutocomplete  onSelect = {handleSelect} value={address} onChange={setAddress}>
+                              {({getInputProps,suggestions,getSuggestionItemProps,loading})=>(
+                                  <>
+                                        <input type="text" className="form-control" name="street"
+                                              id="street"
+                                                {...getInputProps({placeholder:"Type Address"})}
+                                                 ref={register}
+                                                />
+
+                                           {/* No spinners are set here as of now. You can re-use the loader from EventContainer page; 
+                                           though the size of the spinner is set as 10em,fixed in main.scss file. */}
+                                           {loading? 'Loading...':null}
+                                          
+                                         { suggestions.length>0 && <div data-testid="suggestions" className="suggestions-container">{suggestions.map((suggestion)=>{
 
-                            {errors.zip && (
-                                <span className="validationError">{errors.zip}</span>
-                            )}
+                                            
+                                            return <div {...getSuggestionItemProps(suggestion)} key={suggestion.id}>{suggestion.description}</div>
+                                          })}
+                                          </div>
+                                      }
+                                    </>
+                               )}
+                              </PlacesAutocomplete>
+
+                        </div> 
+            }
+                        <div className={showAddress?"form-group": "form-group zip-code"}>
+                            <label htmlFor="zip_code">Zip</label>
+                            <input type="text" className="form-control zip" id="zip_code" 
+                            name="zip_code" defaultValue={zip} 
+                            onChange={(e)=> e.target.value.length>4?setShowAddress(true):setShowAddress(false)}
```

Comment:
Is the intention of this to be a zip only input, then changed to address and zip after 4 characters?

Reply:
Good spot and Yes. thats how it works.

Reply:
Hey @seanlawrenz yes. The general idea is that we input the zip code, then once we have a zip code we have an additional option of an address with auto-complete assist. That process is all to get to a user's lat/long which is then fed into the agencies API request. That triggers the logic that then shows the cards sorted by distance from the user's address to the location.